File: 112.LeetCode@53_Maximum_Subarray/112.LeetCode@53_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        max_sum = nums[0]
        curr_sum = 0

        for num in nums:
            curr_sum += num
            max_sum = curr_sum if curr_sum > max_sum else max_sum
            if curr_sum < 0:
                curr_sum = 0

        return max_sum

File: 112.LeetCode@53_Maximum_Subarray/112.LeetCode@53_Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0], currSum = 0;

        for (int i : nums) {
            currSum += i;
            maxSum = currSum > maxSum ? currSum : maxSum;
            if (currSum < 0) currSum = 0;
        }

        return maxSum;
    }
}